File: data/src/main/java/com/ahn/data/currency/repository/CurrencySelectionRepositoryImpl.kt
package com.ahn.data.currency.repository

import com.ahn.data.currency.local.datasource.CurrencySelectionDataSource
import com.ahn.domain.currency.repository.CurrencySelectionRepository
import javax.inject.Inject

class CurrencySelectionRepositoryImpl @Inject constructor(private val dataSource: CurrencySelectionDataSource) :
    CurrencySelectionRepository {
    override suspend fun getCalculatorSelection() = dataSource.getCalculatorSelection()

    override suspend fun saveCalculatorMainCurrencyCode(code: String) {
        dataSource.saveCalculatorMainCurrencyCode(code)
    }

    override suspend fun saveCalculatorSubCurrencyCode(code: String) {
        dataSource.saveCalculatorSubCurrencyCode(code)
    }

    override suspend fun saveCalculatorSelection(mainCode: String, subCode: String) {
        dataSource.saveCalculatorSelection(mainCode, subCode)
    }

    override suspend fun getExchangeSelection() = dataSource.getExchangeSelection()

    override suspend fun saveExchangeFromCurrencyCode(code: String) {
        dataSource.saveExchangeFromCurrencyCode(code)
    }

    override suspend fun saveExchangeToCurrencyCode(code: String) {
        dataSource.saveExchangeToCurrencyCode(code)
    }

    override suspend fun saveExchangeSelection(fromCode: String, toCode: String) {
        dataSource.saveExchangeSelection(fromCode, toCode)
    }
}

File: ios/Packages/Data/Sources/Data/Currency/Repository/CurrencySelectionRepositoryImpl.swift
import Domain

public struct CurrencySelectionRepositoryImpl: CurrencySelectionRepository {
    private let dataSource: CurrencySelectionDataSource

    public init(dataSource: CurrencySelectionDataSource) {
        self.dataSource = dataSource
    }

    public func calculatorSelection() async throws -> CalculatorCurrencySelection {
        await dataSource.calculatorSelection()
    }

    public func saveCalculatorMainCurrencyCode(_ code: String) async throws {
        await dataSource.saveCalculatorMainCurrencyCode(code)
    }

    public func saveCalculatorSubCurrencyCode(_ code: String) async throws {
        await dataSource.saveCalculatorSubCurrencyCode(code)
    }

    public func saveCalculatorSelection(mainCode: String, subCode: String) async throws {
        await dataSource.saveCalculatorSelection(mainCode: mainCode, subCode: subCode)
    }

    public func exchangeSelection() async throws -> ExchangeCurrencySelection {
        await dataSource.exchangeSelection()
    }

    public func saveExchangeFromCurrencyCode(_ code: String) async throws {
        await dataSource.saveExchangeFromCurrencyCode(code)
    }

    public func saveExchangeToCurrencyCode(_ code: String) async throws {
        await dataSource.saveExchangeToCurrencyCode(code)
    }

    public func saveExchangeSelection(fromCode: String, toCode: String) async throws {
        await dataSource.saveExchangeSelection(fromCode: fromCode, toCode: toCode)
    }
}
